src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c

```diff
@@ -30,17 +31,93 @@ const ActorInit Bg_Ingate_InitVars = {
     (ActorFunc)BgIngate_Update,
     (ActorFunc)BgIngate_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ingate/BgIngate_SetupAction.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ingate/BgIngate_Init.s")
+extern UNK_TYPE D_060011B8;
+extern Gfx D_06001040[];
+
+void BgIngate_SetupAction(BgIngate* this, BgIngateActionFunc func) {
+    this->actionFunc = func;
+}
+
+void BgIngate_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgIngate* this = THIS;
+    s32 pad;
+    s32 sp32 = 0;
+
+    DynaPolyInfo_SetActorMove(&this->dyna, 0);
+    DynaPolyInfo_Alloc(&D_060011B8, &sp32);
+
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, sp32);
+
+    if ((globalCtx->sceneNum != 0x63 || gSaveContext.linkAge != 0) ||
```

Comment:
we have an enum for scenes so you can find the name for 0x63
and age != 0 is LINK_IS_ADULT

Reply:
This one was LINK_IS_CHILD actually